core/lib/multivm/src/versions/vm_latest/bootloader_state/utils.rs

```diff
@@ -116,10 +116,19 @@ pub(crate) fn apply_pubdata_to_memory(
     memory: &mut BootloaderMemory,
     pubdata_information: PubdataInput,
 ) {
+    // Skipping two slots as they will be filled by the bootloader itself:
+    // - One slot is for the selector of the call to the L1Messenger.
+    // - The other slot is for the 0x20 offset for the calldata.
     let l1_messenger_pubdata_start_slot = OPERATOR_PROVIDED_L1_MESSENGER_PUBDATA_OFFSET + 2;
 
-    pubdata_information
-        .build_pubdata()
+    let pubdata = pubdata_information.build_pubdata();
+
+    assert!(
+        pubdata.len() / 32 <= OPERATOR_PROVIDED_L1_MESSENGER_PUBDATA_SLOTS - 2,
```

Comment:
nit: this assert is a little bit incorrect - if pubdata.len is not divisible by 32.

Reply:
it is enforced by `build_pubdata` that it is divisible by 32